### PR TITLE
Add wait condition for training set's dependent features and label

### PR DESCRIPTION
**Pull request by Sami1309**
# Description

Solves issue where training set attempts to execute without waiting for its dependent features/labels to be ready

### Select type(s) of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

# Checklist:

- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I have fixed any merge conflicts

### CODE REVIEW

File: coordinator/coordinator.go [hunk 1]
```diff
@@ -119,6 +119,50 @@ func (c *Coordinator) AwaitPendingSource(sourceNameVariant metadata.NameVariant)
 	return nil, fmt.Errorf("waited too long for source to become ready")
 }
 
+func (c *Coordinator) AwaitPendingFeature(featureNameVariant metadata.NameVariant) (*metadata.FeatureVariant, error) {
+	featureStatus := metadata.PENDING
+	start := time.Now()
+	elapsed := time.Since(start)
+	for featureStatus != metadata.READY && elapsed < time.Duration(c.Timeout)*time.Second {
```
Comment (sdreyer) [inline coordinator/coordinator.go:126]: Remove timeouts

**Review by sdreyer — APPROVED** [review]
Remove timeouts then lgtm

File: coordinator/coordinator.go [hunk 2]
```diff
@@ -113,10 +110,47 @@ func (c *Coordinator) AwaitPendingSource(sourceNameVariant metadata.NameVariant)
...
+func (c *Coordinator) AwaitPendingFeature(featureNameVariant metadata.NameVariant) (*metadata.FeatureVariant, error) {
+	featureStatus := metadata.PENDING
+	for featureStatus != metadata.READY {
+		feature, err := c.Metadata.GetFeatureVariant(context.Background(), featureNameVariant)
+		if err != nil {
+			return nil, err
+		}
+		featureStatus := feature.Status()
+		if featureStatus == metadata.FAILED {
+			return nil, fmt.Errorf("source of feature not ready: name: %s, variant: %s", featureNameVariant.Name, featureNameVariant.Variant)
```
Comment (sdreyer) [inline coordinator/coordinator.go:127]: Can you change error message to "feature registration failed: ......". Same with the label and source. Want it to be explicit that its in a failed state

File: coordinator/coordinator.go (hunk 1 above)
Comment (Sami1309) [reply coordinator/coordinator.go]: removed

**Pull request merged**